app/src/main/java/org/mozilla/focus/observer/AverageLoadTimeObserver.java

```diff
@@ -24,8 +31,7 @@ protected void onValueChanged(Boolean loading) {
             }
         } else {
             if (loadStarted) {
-                Log.i(LOG_TAG, "Loaded page at " +
-                        SessionManager.getInstance().getCurrentSession().getUrl().getValue());
+                Log.i(LOG_TAG, "Loaded page at " + session.getUrl().getValue());
```

Comment:
We don't need to null check session here?

Reply:
We registered this observer on a property of the session object in BrowserFragment. So for the current version of the code the reference can't be null when AverageLoadTimeObserver is created (and we never clear it inside the observer).

But yeah, I think I'm going to add @NonNull to the constructor to make this more clear and avoid mistakes in the future.